projects/storefrontlib/src/cms-structure/outlet/outlet.directive.ts

```diff
@@ -21,15 +37,69 @@ export class OutletDirective implements OnInit {
     this._context = value;
   }
 
+  /**
+   * Defers loading options for the the templates of this outlet.
+   */
+  @Input() cxOutletDefer: IntersectionOptions = {};
+
+  constructor(
+    vcr: ViewContainerRef,
+    templateRef: TemplateRef<any>,
+    outletService: OutletService<TemplateRef<any> | ComponentFactory<any>>,
+    // tslint:disable-next-line: unified-signatures
+    intersectionService?: IntersectionService
+  );
+  /**
+   * @deprecated since version 1.4
+   * Use constructor(vcr: ViewContainerRef, templateRef: TemplateRef<any>, outletService: OutletService<TemplateRef<any> | ComponentFactory<any>>, intersectionService?: IntersectionService) instead
+   */
+  constructor(
+    vcr: ViewContainerRef,
+    templateRef: TemplateRef<any>,
+    outletService: OutletService<TemplateRef<any> | ComponentFactory<any>>
+  );
   constructor(
     private vcr: ViewContainerRef,
     private templateRef: TemplateRef<any>,
     private outletService: OutletService<
       TemplateRef<any> | ComponentFactory<any>
-    >
+    >,
+    private intersectionService?: IntersectionService
   ) {}
 
   ngOnInit(): void {
+    this.renderInstantly() ? this.render() : this.deferRender();
+  }
+
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
+  }
+
+  private renderInstantly(): boolean {
+    return (
+      !this.intersectionService ||
+      (this.cxOutletDefer.deferLoading &&
+        this.cxOutletDefer.deferLoading === DeferLoadingStrategy.INSTANT)
+    );
+  }
+
+  private deferRender() {
+    const hostElement = this.getHostElement(this.vcr.element.nativeElement);
+    this.addPendingStyle(hostElement);
+
+    // we need to wait a tick to take advantage of the ghost layout
+    setTimeout(() => {}, 0);
+    this.subscription.add(
```

Comment:
We can probably get rid of that subscription, because observable will emit only once and complete (so no need to unsubscribe).